taichi/common/interface.h

```diff
@@ -308,28 +308,7 @@ class InterfaceHolder {
           new TI_IMPLEMENTATION_HOLDER_NAME(class_name)(base_alias);          \
     }                                                                         \
     return TI_IMPLEMENTATION_HOLDER_PTR(class_name);                          \
-  }                                                                           \
-  class InterfaceInjector_##class_name {                                      \
```

Comment:
Sounds good. At a high level, I feel like the whole `InterfaceHolder`/`InterfaceInjector` part is a bit over-engineered now. While it served the old Taichi *library* well, after we switch to a compiler no longer widely use it.

Maybe it worth considering refactoring, e.g., different backends using a similar interface/impl separation, if that makes sense. In that case, we still need the classes here, possibly with some refactoring.

Reply:
I see. Since I was mostly shuffling the code, I didn't really look very deep into what InterfaceHolder/InterfaceInjector is actually doing... Yeah it'll be great if we can have a common JIT interface for different backends.